src/LanguageServer/IdePurescript/Imports.purs

```diff
@@ -63,6 +66,11 @@ addCompletionImport log docs config state args = do
       { moduleName: mod', qualifier: Just qual'} -> mod == mod' && qual == qual'
       _ -> false
 
+    addOpenImport modules port uri text mod =
+      addModuleImport modules port uri text mod <#> case _ of
+        Just r -> r { result = UpdatedImports r.result }
+        Nothing -> { state: modules, result: FailedImport }
```

Comment:
This is a little weird because it's undoing what addModuleImport does.

Reply:
Oh I see, just massaging the data back and forth. Maybe a comment then?

Reply:
Yeah, `addModuleImport` discards the result data type and wraps it in Maybe, and we have to add it back for the types to unify.